### PR TITLE
bleat.startScan updates

### LINKED ISSUE #44: Support for reportning devices multiple times during scanning
The Bleat Classing API reports devices once when found. In for instance beacon applications devices need to be reported multiple times. Can we add support for this? Like a flag, function/setting to control scanning behaviour. I am happy to help with this.

Did a quick experiment, commented out the following line in api.classic.js:

    adapter.startScan(serviceUUIDs, function(deviceInfo) {
        var device = new Device(deviceInfo);
        //if (devices[device.address]) return;
        devices[device.address] = device;
        if (foundFn) foundFn(device);

This makes devices being reported multiple times.

One potential problem is if the application expects the device object to be the same between calls to foundFn. For example, the application might add custom properties to the device object. These would be lost with the above code. Properties need to be copied to prevent this problem.

**Edit:** Been thinking more about the idea of a "persistent" device object that has the same identity between calls to foundFn. Probably this is a bad idea, for example, if device sends multiple packets you would want a guarantee that the data is not overwritten by a concurrent operation. Now, JavaScript is single-threaded, but there may be Worker objects etc, so this could happen. Eddystone is an example of sending multiple packets. 

We used this technique ("persistent" device objects) in the Evothings EasyBLE library, but now I realize it is perhaps not the best solution. It would be better to have the application handle this, and this would also make for a cleaner separation between "model and view", so to speak.

### LINKED ISSUE #46: bleat.startScan: Callback functions not properly referenced when leaving out param serviceUUIDs
When leaving put parameter serviceUUIDs in function bleat.startScan, foundFn is set properly, but the other callbacks are not updated and reference wrong functions.

In this example, foundFn will be called properly, but errorFn will be called in place of completeFn, and errorFn will be undefined inside startScan:

    bleat.startScan(foundFn, completeFn, errorFn);

### PR DESCRIPTION
**Pull request by mikaelkindborg**
Fixed callback function references in bleat.startScan, fixes #46 

Added options object to bleat.startScan with option allowDuplicates, resolves #44

### CODE REVIEW

**Comment by thegecko** [comment]
Could you split this into 2 separate PRs fixing #44 and #46 individually?

File: dist/api.classic.js [hunk 1]
```diff
@@ -250,8 +250,12 @@
             adapters[adapterName] = definition;
             adapter = definition;
         },
-        startScan: function(serviceUUIDs, foundFn, completeFn, errorFn) {
+        startScan: function(serviceUUIDs, foundFn, completeFn, errorFn, options) {
```
Comment (thegecko) [inline dist/api.classic.js]: IMO an options object should be used to replace all args. Can you stick with arguments for this addition and just have an allowDuplicates argument?

File: dist/api.classic.js [hunk 2]
```diff
@@ -261,7 +265,8 @@
             var devices = {};
             adapter.startScan(serviceUUIDs, function(deviceInfo) {
                 var device = new Device(deviceInfo);
-                if (devices[device.address]) return;
+                var allowDuplicates = options && options.allowDuplicates;
+                if (devices[device.address] && !allowDuplicates) return;
```
Comment (thegecko) [inline dist/api.classic.js]: Without options, this becomes:
```javascript
if (devices[device.address] && returnSingle) return;
```

File: dist/api.classic.js (hunk 1 above)
Comment (mikaelkindborg) [inline dist/api.classic.js]: Thanks, great will update the pull request!

**Comment by mikaelkindborg** [comment]
@thegecko What is the best method for splitting into two pull requests?

**Comment by thegecko** [comment]
@mikaelkindborg Nevermind :)

**Comment by mikaelkindborg** [comment]
@thegecko Thanks for merging!

**Pull request merged**